genai-perf/genai_perf/parser.py

```diff
@@ -1493,8 +1140,12 @@ def parse_args():
             return args, None, argv[passthrough_index + 1 :]
         else:
             # For all other subcommands, parse the CLI fully (no config file)
-            config = ConfigCommand({"model_name": args.formatted_model_name})
+            config = ConfigCommand(
+                {"model_names": args.model}, skip_inferencing_and_checking=True
+            )
             config = add_cli_options_to_config(config, args)
+            config.infer_and_check_options()
```

Comment:
This flow is a bit confusing. The code initializes the ConfigCommand with skip_inferencing_and_checking=True, then it calls it right after. Is there a way to clean up this flow to allow the cli_options to be used as an optional init parameter, so that these lines just becomes
`config=ConfigCommand(args)` or something similar?

Reply:
This is because we need to delay inferencing/checking until after we have assigned values to the config file via the `add_cli_options...` method.

The only other option is to make the `add_cli_options` method internal to the Config class, which I am reluctant to do.